src/slurm_plugin/slurm_resources.py

```diff
@@ -231,6 +229,38 @@ def is_poweing_up_idle(self):
     def is_ice(self):
         return self.error_code in self.EC2_ICE_ERROR_CODES
 
+    def is_reboot_requested(self):
+        return self.SLURM_SCONTROL_REBOOT_REQUESTED_STATE in self.states
+
+    def is_reboot_issued(self):
+        return self.SLURM_SCONTROL_REBOOT_ISSUED_STATE in self.states
+
+    def is_rebooting(self):
+        """
+        Check if the node is rebooting via scontrol reboot.
+
+        Check that the node is in a state consistent with the scontrol reboot request, which could trigger
+        the health checks.
+        """
+        cond = False
+        if self._is_drain():
```

Comment:
We want to consider any nodes with `REBOOT_ISSUED+DRAIN` to be rebooting nodes, right?

Reply:
Correct. and in the line below it's what we do (247 in that commit).